cmd/fleet/serve.go

```diff
@@ -553,17 +553,19 @@ the way that the Fleet server works.
 			var (
 				wstepCertManager microsoft_mdm.CertManager
 			)
-			// TODO: check if MicrosoftMDM is enabled?
-			if config.MDM.IsMicrosoftWSTEPSet() {
-				_, crtPEM, keyPEM, err := config.MDM.MicrosoftWSTEP()
-				if err != nil {
-					initFatal(err, "validate Microsoft WSTEP certificate and key")
-				}
-				wstepCertManager, err = microsoft_mdm.NewCertManager(ds, crtPEM, keyPEM)
-				if err != nil {
-					initFatal(err, "initialize mdm microsoft wstep depot")
-				}
 
+			if configpkg.IsMDMFeatureFlagEnabled() && appCfg.MDM.WindowsEnabledAndConfigured {
```

Comment:
Is it a problem that `appCfg.MDM.WindowsEnabledAndConfigured` would be `false` by default?

I'm thinking about scenarios where Windows MDM has never been turned on. In that case, wouldn't the user first have to turn on Windows MDM in the UI and then restart the server so that the CA can be initialized?

Reply:
For Apple, I believe we're taking a different approach where the `appCfg.MDM.EnabledAndConfigured` depends on whether the certs are set (see line 509 above).

Reply:
Good point, I've modified the logic to only check for the feature flag. As you mentioned, `appCfg.MDM.WindowsEnabledAndConfigured` will depend on runtime appconfig being set, and `config.MDM.IsMicrosoftWSTEPSet` is more of a start-time setting, so only the feature flag check should be done here.